tests/native_tests/dot.hy

```diff
@@ -0,0 +1,20 @@
+(defclass X [object] [])
```

Comment:
There are already some tests for dot application [here](https://github.com/Kodiologist/hy/blob/821ed96faf0dd6467307759ec07ab444de29e5e8/tests/native_tests/language.hy#L399); try adding these there instead.